src/client-app/src/models/Pieces/Helper/MoveHelper.ts

```diff
@@ -0,0 +1,31 @@
+import { Game } from "../../game";
+import { Coordinate } from "../../chess";
+
+export class MoveHelper {
+  //Checks for unobstructed continuos empty cells in one directions as valid moves
+  //stops if a cell contains a piece and returns valid moves upto and including that point
+  static checkDirection(gameState: Game, row: number, col: number, dx: number, dy: number): Coordinate[] {
+    //base case
+    let moves: Coordinate[] = [];
+    const newRow = row + dx;
+    const newCol = col + dy;
+
+    if (newRow >= 0 && newRow <= 7 && newCol >= 0 && newCol <= 7) {
```

Comment:
Can we extract this to another helper function for reusability?

Reply:
which part did you want pulled out? I am not clear on this one

Reply:
Ah this one sry (newRow >= 0 && newRow <= 7 && newCol >= 0 && newCol <= 7)

Reply:
sure, got it